java_src/foam/nanos/auth/AuthOAO.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.auth;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.oao.ProxyOAO;
+
+import javax.security.auth.AuthPermission;
+import java.util.Map;
+
+public class AuthOAO
+  extends ProxyOAO
+{
+  private final String rootPermission;
+
+  public AuthOAO() {
+    this.rootPermission = getClass().getSimpleName();
+  }
+
+  public AuthOAO(String rootPermission) {
+    this.rootPermission = rootPermission;
+  }
+
+  @Override
+  public FObject get(X x) throws IllegalAccessException {
+    AuthService authService = (AuthService) x.get("authService");
+    java.security.Permission permission = new AuthPermission(rootPermission + "." + getDelegate().get(x).getClassInfo().getId());
```

Comment:
I think this should just be 'rootPermission' and 'rootPermission' should default to getDelegate().get(x).getClassInfo().getId() if not explicitly set.